### ISSUE TITLE
Footnotes not rendering as expected in localized version

### ISSUE THREAD

**Issue by thorge** [executablebooks/MyST-Parser#801]
### What version of `myst-parser` are you using?

2.0.0

### What version dependencies are you using?
sphinx-build (7.2.6)
sphinx-rtd-theme
sphinx-design
sphinxcontrib-mermaid
sphinxcontrib-svg2pdfconverter
sphinx-intl

### What operating system are you using?

Linux

### Describe the Bug

I have a Sphinx documentation using *myst* and *sphinx-intl*. In the default English version of the documentation, footnotes are formatted correctly, and I'm getting properly rendered footnotes in the resulting HTML when I build the documentation.

However, I've also created a localized German version of the documentation, and while the translation seems to work fine overall, I'm encountering an issue with footnotes. In the German version, the footnotes are not being converted to the correct HTML format. Instead of seeing the formatted footnotes as in the English version, it appears that the German version is simply printing the Markdown string, like `[^1]`, without converting it to the expected HTML footnote markup. The footnote section works fine, but the inline link is not formatted and linked correctly.

```sh
/docs/source/test.md:1: WARNING: inconsistent footnote references in translated message. original: ['[^1]'], translated: []                                                                                                                   
```

![test](https://github.com/executablebooks/MyST-Parser/assets/4233305/79ac409a-b38f-4eb1-b6e3-8bd9957e4b24)

The .po files look fine, e.g.:

```
#: ../test.md:1
msgid ""
"Test[^1]"
msgstr ""
"Test[^1]"
```

### Expected Behavior

I expect that when I build the German version of the documentation, the footnotes should be processed and rendered in the same way as they are in the original English version. Other markdown features also work without interfering with the locales.

**Comment by j123b567** [comment]
Probably duplicate of #445

**Comment by chrisjsewell** [comment]
> Probably duplicate of #445

Doesn't matter, it's fixed in #931 anyway 😉
